subsys/bluetooth/mesh/transport.c

```diff
@@ -992,31 +992,106 @@ int bt_mesh_ctl_send(struct bt_mesh_net_tx *tx, u8_t ctl_op, void *data,
 	       tx->ctx->addr, tx->ctx->send_ttl, ctl_op);
 	BT_DBG("len %zu: %s", data_len, bt_hex(data, data_len));
 
-	buf = bt_mesh_adv_create(BT_MESH_ADV_DATA, tx->xmit, BUF_TIMEOUT);
-	if (!buf) {
-		BT_ERR("Out of transport buffers");
-		return -ENOBUFS;
+	if (data_len <= 11) {
+		buf = bt_mesh_adv_create(BT_MESH_ADV_DATA,
+					 tx->xmit, BUF_TIMEOUT);
+		if (!buf) {
+			BT_ERR("Out of transport buffers");
+			return -ENOBUFS;
+		}
+
+		net_buf_reserve(buf, BT_MESH_NET_HDR_LEN);
+
+		net_buf_add_u8(buf, TRANS_CTL_HDR(ctl_op, 0));
+
+		net_buf_add_mem(buf, data, data_len);
+		if (IS_ENABLED(CONFIG_BT_MESH_FRIEND)) {
+			if (bt_mesh_friend_enqueue_tx(tx, BT_MESH_FRIEND_PDU_SINGLE,
+				seq_auth, &buf->b) && BT_MESH_ADDR_IS_UNICAST(tx->ctx->addr)) {
```

Comment:
The segment count parameter is missing..Should it be 1?
tx doesn't have a seg_n (tx->seg_n)